modules/core/site.js

```diff
@@ -58,6 +58,18 @@ $.fn.fadeOutAndRemove = function(timeout = 600) {
     return this;
 };
 
+$.fn.modal = function(action) {
```

Comment:
Hello @Shadow243 
Shouldn't we mix this with Hm_Modal ?

Reply:
I thought about this idea then I actually left modal to reflect the behavior of bootstrap especially since it is an implementation of Bootstrap modal

Reply:
@josaphatim  If I use JS to generate dynamically the modal I will then not be able to write phpunit tests, So I favored the code that I can easily test.